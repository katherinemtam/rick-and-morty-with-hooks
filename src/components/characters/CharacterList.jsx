import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import { Link } from 'react-router-dom';

const CharacterList = ({ characters }) => (
  <ul>
    {characters.map(character => (
      <Link to={`/${character.id}`} key={character.id}>
        <li>
          <Character 
            name={character.name}
            image={character.image}
          />
        </li>
      </Link>
    ))}
  </ul>
);



CharacterList.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.int,
      name: PropTypes.string,
      status: PropTypes.string,
      species: PropTypes.string,
      type: PropTypes.string,
      gender: PropTypes.string,
      origin: PropTypes.object,
      location: PropTypes.object,
      image: PropTypes.string,
      episode: PropTypes.arrayOf(
        PropTypes.string
      ),
    })
  ).isRequired,
};

export default CharacterList;
