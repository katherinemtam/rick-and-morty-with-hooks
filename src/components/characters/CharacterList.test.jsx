import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  const character = [
    {
      name: 'Pickle Rick',
      image: 'http://image.com'
    }
  ];
  
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(<MemoryRouter><CharacterList characters={character}/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
