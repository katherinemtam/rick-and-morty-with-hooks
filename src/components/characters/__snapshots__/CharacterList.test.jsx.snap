// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul>
    <a
      href="/undefined"
    >
      <li>
        <h2>
          Pickle Rick
        </h2>
        <img
          src="http://image.com"
        />
      </li>
    </a>
  </ul>
</DocumentFragment>
`;
